lib/core/src/refund/chain.rs

```diff
@@ -0,0 +1,273 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use boltz_client::{
+    boltz::{BOLTZ_MAINNET_URL_V2, BOLTZ_TESTNET_URL_V2},
+    network::electrum::ElectrumConfig,
+    util::secrets::Preimage,
+    Amount, BtcSwapTx,
+};
+use log::info;
+use tokio::sync::Mutex;
+
+use crate::{
+    chain::{bitcoin::BitcoinChainService, liquid::LiquidChainService},
+    error::{PaymentError, SdkResult},
+    prelude::{ChainSwap, Config, Direction, LiquidNetwork, SwapScriptV2},
+    swapper::BoltzSwapper,
+};
+
+use super::{
+    common::{self, get_lbtc_fee_rate_sats_per_vbyte},
+    RefundHandler,
+};
+
+pub(crate) struct ChainRefundHandler<SwapperType: ?Sized> {
+    config: Config,
+    swapper: Option<Arc<SwapperType>>,
+    liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
+    bitcoin_chain_service: Arc<Mutex<dyn BitcoinChainService>>,
+}
+
+impl ChainRefundHandler<BoltzSwapper> {
+    pub(crate) fn new(
+        config: Config,
+        swapper: Arc<BoltzSwapper>,
+        liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
+        bitcoin_chain_service: Arc<Mutex<dyn BitcoinChainService>>,
+    ) -> Self {
+        Self {
+            config,
+            swapper: Some(swapper),
+            liquid_chain_service,
+            bitcoin_chain_service,
+        }
+    }
+
+    fn boltz_url(&self) -> &str {
```

Comment:
Given [your answer above](https://github.com/breez/breez-sdk-liquid/pull/407/files#r1758877086), `boltz_url` should IMO either return the swapper value or throw an error.

For `boltz_url`, the swapper includes logic that builds a special `boltz_url` with the referral ID as GET argument. See `BoltzSwapper::new` and #329. By building a separate `boltz_url` here (outside of the swapper) and bypassing the swapper logic, that `boltz_url` will not have the referral ID.

It also makes sense intuitively: if no swapper (Boltz API) is used for this instance of the refund handler, then there is no need for a `boltz_url`.